OctopusDSC/OctopusDSCHelpers.ps1

```diff
@@ -158,41 +158,45 @@ Function Get-MaskedOutput
     [CmdletBinding()]
     param($arguments)
 
-    $reg = [System.Text.RegularExpressions.RegEx]::new("--masterkey|--password|--license",
-                [System.Text.RegularExpressions.RegexOptions]::IgnoreCase)
+    $singleAsterixArgs = "--masterkey|--license|--trust|--password|--remove-trust|--apikey|--pw|--pfx-password|--proxyPassword";
+    $connectionStringArgs = "--connectionstring";
 
-    if(($arguments -match "--masterkey|--password|--license"))
-    {
-        for($x=0;$x -lt $arguments.count; $x++)
-        {
-            if(($arguments[$x] -match "--masterkey|--password|--license|--trust|--remove-trust|--apikey|--password|--pw|--pfx-password|--proxyPassword"))
-            {
-                $arguments[$x+1] = $arguments[$x+1] -replace ".", "*"
-            }
-        }
-        $out = $arguments
-    }
-    elseif(($arguments -match "password|pwd"))
+    $combinedArgs = Join-String -Separator "|" -InputObject $singleAsterixArgs, $connectionStringArgs
+
+    # early bail for edge case where many args are passed
+    if (!$arguments -match $combinedArgs)
     {
-        $out = $arguments -replace "(password|pwd)=[^;|`"]*", "`$1=********"
+        return @("************************")
     }
-    else
+
+    # Scrub sensitive values
+    for($x=0; $x -lt $arguments.count; $x++)
     {
-        $out = @("************************")
+        if($arguments[$x] -match $singleAsterixArgs)
+        {
+            $arguments[$x+1] = $arguments[$x+1] -replace ".", "*"
+        }
+
+        if($arguments[$x] -match $connectionStringArgs)
+        {
+            $arguments[$x+1] = $arguments[$x+1] -replace "(password|pwd)=[^;|`"]*", "`$1=********"
+        }
     }
-    return $out
+
+    return $arguments
+}
+
+function Write-MaskedCommandVerbose ($cmdArgs) {
+    $copiedarguments = @() # hack to pass a copy of the array, not a reference
+    $copiedarguments += $cmdArgs
+    $maskedarguments = Get-MaskedOutput $copiedarguments
+    Write-Verbose "Executing command '$octopusServerExePath $($maskedarguments -join ' ')'"
 }
 
 function Invoke-OctopusServerCommand ($cmdArgs) {
-    # todo: fix this up
```

Comment:
Rewriting the function masker helped make it clear that much of this code was redundant, so it has been removed. Args provided to a call to Write-Verbose (or other logging) should always be passed through the masking function.